src/services/PostEventProcessor.ts

```diff
@@ -112,12 +113,20 @@ export class PostEventProcessor {
 					break;
 				}
 				case Constants.F2F_YOTI_START: {
-					updateExpression = "SET journeyWentAsyncOn = :journeyWentAsyncOn, expiresOn = :expiresOn";
+					const fetchedRecord = await this.iprServiceAuth.getAuthEventBySub(userId);
+					if (!fetchedRecord) {
+						this.logger.error({ message: "F2F_YOTI_START event sent before AUTH_IPV_AUTHORISATION_REQUESTED event." }, { messageCode: MessageCodes.SQS_OUT_OF_SYNC });
+						throw new AppError(HttpCodesEnum.BAD_REQUEST, Constants.SQS_OUT_OF_SYNC);
```

Comment:
throw a Server error here and error message should not be a messageCode, use the message used in the logger in the above line.